### ISSUE TITLE
[BUG] Invalid next URL when session expired

### ISSUE THREAD

**Issue by Hailong-am** [opensearch-project/security-dashboards-plugin#2135]
**What is the bug?**

Here is the url with error 

http://localhost:5601/app/login?nextUrl=%2Fapp%2Fopensearch_index_management_dashboards_%252Frollups#/rollups?from=0&search=&size=20&sortDirection=desc&sortField=_id

```json
{
    "statusCode": 400,
    "error": "Bad Request",
    "message": "[request query.nextUrl]: Invalid nextUrl parameter."
}
```
The nextUrl is `/app/opensearch_index_management_dashboards_%2Frollups`
Based on the validation rule, it has `%2F` which is not allowed.

https://github.com/opensearch-project/security-dashboards-plugin/blob/506d803b868002f131a65d0d7ad370625454a8e4/server/utils/next_url.ts#L75-L80

**How can one reproduce the bug?**
1. Enable workspace and goes into a workspace
2. Go to http://localhost:5601/app/opensearch_index_management_dashboards_%2Frollups#/rollups?dataSourceId=&from=0&search=&size=20&sortDirection=desc&sortField=_id
3. Wait for session expired

**What is the expected behavior?**

**What is your host/environment?**
 - Plugins

**Do you have any screenshots?**

**Do you have any additional context?**

**Comment by cwperks** [comment]
@Hailong-am Why is the URL http://localhost:5601/app/login?nextUrl=%2Fapp%2Fopensearch_index_management_dashboards_%252Frollups#/rollups?from=0&search=&size=20&sortDirection=desc&sortField=_id instead of http://localhost:5601/app/login?nextUrl=%2Fapp%2Fopensearch_index_management_dashboards_%2Frollups#/rollups?from=0&search=&size=20&sortDirection=desc&sortField=_id

i.e. Why the `%25`. IMO the logic for nextUrl validation is sound, but the url you have pasted is doubly-encoded.

**Comment by cwperks** [comment]
@Hailong-am Is `encodeUriComponent` required when declaring the paths in the index-management-dashboards-plugin [here](https://github.com/opensearch-project/index-management-dashboards-plugin/blob/64014ef9ac79566824f6003001506020575a8844/public/plugin.ts#L184)?

**Comment by Hailong-am** [comment]
> @Hailong-am Is `encodeUriComponent` required when declaring the paths in the index-management-dashboards-plugin [here](https://github.com/opensearch-project/index-management-dashboards-plugin/blob/64014ef9ac79566824f6003001506020575a8844/public/plugin.ts#L184)?

I think so, the application id `opensearch_index_management_dashboards_${encodeURIComponent('/routes')}`, if remove the encodeURIComponent the application id will be `opensearch_index_management_dashboards_/routes`, so url will be `/app/opensearch_index_management_dashboards_/routes/...` that will render application with id `opensearch_index_management_dashboards_` which is not exists

**Comment by Hailong-am** [comment]
> route ids are explicitly defined and not based on the route configured. Examples: https://github.com/opensearch-project/security-dashboards-plugin/blob/main/public/plugin.ts#L189-L275

yes, that's the cause. application id could not contains any special character like `/` `+`, otherwise it will have same problem. 

either we have a place to document this limitation, or we figure out a way to check the application id is a valid and exist one no matter what kind of format does it have.
